diff --git a/env.c b/env.c
--- a/env.c
+++ b/env.c
@@ -61,7 +61,7 @@
 /**
    Command used to start fishd
 */
-#define FISHD_CMD L"fishd ^/tmp/fishd.log.%s"
+#define FISHD_CMD L"@GENTOO_PORTAGE_EPREFIX@/bin/fishd ^@GENTOO_PORTAGE_EPREFIX@/var/tmp/fishd.log.%s"
 
 /**
    Value denoting a null string
diff --git a/share/config.fish.in b/share/config.fish.in
--- a/share/config.fish.in
+++ b/share/config.fish.in
@@ -6,6 +6,15 @@
 # @configure_input@
 
 #
+# Load environment settings from profile.env,  which is created by
+# env-update from the files in /etc/env.d
+#
+
+if test -e @GENTOO_PORTAGE_EPREFIX@/etc/profile.env
+	@GENTOO_PORTAGE_EPREFIX@/bin/sed -e 's/^export \([A-Za-z_]\+\)=\(.*\)$/set -x \1 \2/g' -e 'y/:/ /' -e "y/'/ /" @GENTOO_PORTAGE_EPREFIX@/etc/profile.env | .
+end
+
+#
 # Set default field separators
 #
 
@@ -72,18 +81,21 @@
 # want this even for text-only terminals.
 #
 
-set -l path_list /bin /usr/bin /usr/X11R6/bin /usr/local/bin @prefix@/bin @optbindirs@
+set -l path_list @GENTOO_PORTAGE_EPREFIX@/usr/bin @GENTOO_PORTAGE_EPREFIX@/bin $PATH /usr/bin /bin
 
 # Root should also have the sbin directories in the path
 switch $USER
 	case root
-	set path_list $path_list /sbin /usr/sbin /usr/local/sbin @prefix@/sbin
+	set path_list $path_list @GENTOO_PORTAGE_EPREFIX@/usr/sbin @GENTOO_PORTAGE_EPREFIX@/sbin $ROOT_PATH /usr/sbin /sbin
 end
 
+set -e PATH
+set -e ROOTPATH
+
 for i in $path_list
 	if not contains $i $PATH
 		if test -d $i
-			set PATH $PATH $i
+			set -x PATH $PATH $i
 		end
 	end
 end
